daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -124,72 +124,85 @@ func (l *JSONFileLogger) Log(msg *logger.Message) error {
 }
 
 func writeLog(l *JSONFileLogger) (int64, error) {
+
+	if err := checkCapacityAndRotate(l); err != nil {
+		return -1, err
+	}
+
+	i, err := l.buf.WriteTo(l.f)
+	if err != nil {
+		l.buf.Reset()
```

Comment:
I think buffer just used to made log line(log string) and write it once(Call `WriteTo()` or other writing function). 
i'm not sure container will die or exit for log error(write fail). If don't and buffer(log data) is not flushed when write fail. it' will grow forever.